### PR TITLE
Table: updated Docs to use autogenerated prop table

### PR DESCRIPTION
**Pull request by AlbertCarreras**
#### What changed?

Table: updated Docs to use autogenerated prop table

#### Why?

Gestalt is moving in this direction, all components must autogenerate prop tables

### CODE REVIEW

File: packages/gestalt/src/Table.js [hunk 1]
```diff
@@ -16,18 +16,38 @@ import TableSortableHeaderCell from './TableSortableHeaderCell.js';
...
+  /**
+   * Must be instances of Table.Header, Table.Body, and/or Table.Footer components. See the [Subcomponent section](https://gestalt.pinterest.systems/table#Subcomponents) to learn more.
+   */
   children: Node,
+  /**
+   * Label for screen readers to announce Table.
+   */
   accessibilityLabel: string,
+  /**
+   * Specify a border width for table: "sm" is 1px.
```
Comment (dangerismycat) [inline packages/gestalt/src/Table.js:28]: nit: Table should be capitalized

File: packages/gestalt/src/Table.js [hunk 2]
```diff
@@ -16,18 +16,38 @@ import TableSortableHeaderCell from './TableSortableHeaderCell.js';
...
+   */
   borderStyle?: 'sm' | 'none',
+  /**
+   * Use numbers for pixels: maxHeight={100} and strings for percentages: maxHeight="100%".
```
Comment (dangerismycat) [inline packages/gestalt/src/Table.js:32]: backticks around the example code

File: packages/gestalt/src/Table.js [hunk 3]
```diff
@@ -16,18 +16,38 @@ import TableSortableHeaderCell from './TableSortableHeaderCell.js';
...
+   */
   maxHeight?: number | string,
+  /**
+   * Specify how many columns from the start of the Table should be sticky when scrolling horizontally. See the [sticky column](https://gestalt.pinterest.systems/table#Sticky-Column), the [multiple sticky columns](https://gestalt.pinterest.systems/table#Multiple-sticky-columns), the [sticky header and columns](https://gestalt.pinterest.systems/table#Sticky-header-and-sticky-columns), the [expandable row with sticky columns](https://gestalt.pinterest.systems/table#Table-Row-Expandable-with-Sticky-Columns), and the [sortable header cells with sticky columns](https://gestalt.pinterest.systems/table#Sortable-header-cells-with-sticky-columns) variants for details.
```
Comment (dangerismycat) [inline packages/gestalt/src/Table.js:36]: You can leave off the article after the first list item: "the foo, bar, baz, and foobar variants…"

File: packages/gestalt/src/Table.js [hunk 4]
```diff
@@ -16,18 +16,38 @@ import TableSortableHeaderCell from './TableSortableHeaderCell.js';
...
+   */
   stickyColumns?: ?number,
 |};
 
 /**
- * https://gestalt.pinterest.systems/table
+ * A [Table](https://gestalt.pinterest.systems/table) is a set of structured data that is easy for a user to scan, examine, and compare. Table data is displayed in a grid format and can be used to structure both interactive and static data.
```
Comment (dangerismycat) [inline packages/gestalt/src/Table.js:42]: nit: no article needed, just "Table is a set…"

File: docs/pages/table.js [hunk 5]
```diff
@@ -2718,3 +2552,39 @@ Checkboxes are often used in tables to allow for selecting and editing of multip
...
+      'TableCell',
+      'TableHeaderCell',
+      'TableSortableHeaderCell',
+      'TableRow',
+      'TableRowExpandable',
+    ],
+  });
+
+  docGen.Table.props.children.flowType.raw =
+    'React.ChildrenArray<React.Element< typeof Table.Body | typeof Table.Footer | typeof Table.Header >>';
```
Comment (dangerismycat) [inline docs/pages/table.js:2574]: there's some extra whitespace within the "< >" in these types

File: packages/gestalt/src/TableHeader.js [hunk 6]
```diff
@@ -4,14 +4,21 @@ import cx from 'classnames';
 import styles from './Table.css';
 
 type Props = {|
+  /**
+   * Must be an instance of Table.Row. See the [Subcomponent section](https://gestalt.pinterest.systems/table#Subcomponents) to learn more.
+   */
   children: Node,
+  /**
+   * If true, the table header will be sticky and the table body will be scrollable. See the [sticky Header](https://gestalt.pinterest.systems/table#Sticky-header) and the [sticky header and columns](https://gestalt.pinterest.systems/table#v) variants for details.
```
Comment (dangerismycat) [inline packages/gestalt/src/TableHeader.js:12]: looks like something happened to the "sticky header and columns" link

File: packages/gestalt/src/TableHeaderCell.js [hunk 7]
```diff
@@ -4,28 +4,49 @@ import cx from 'classnames';
...
+  /**
+   * Private prop required for sticky columns
+   */
   shouldBeSticky?: boolean,
+  /**
+   * Private prop required for sticky columns
+   */
   shouldHaveShadow?: boolean,
+  /**
+   * `colSpan` defines the number of columns a cell should span.
```
Comment (dangerismycat) [inline packages/gestalt/src/TableHeaderCell.js:20]: No need to repeat the prop name in its own description

File: packages/gestalt/src/TableHeaderCell.js [hunk 8]
```diff
@@ -4,28 +4,49 @@ import cx from 'classnames';
 import styles from './Table.css';
 
 type Props = {|
+  /**
+   * The content of the table cell.
+   */
   children: Node,
+  /**
+   * Private prop required for sticky columns
```
Comment (dangerismycat) [inline packages/gestalt/src/TableHeaderCell.js:12]: can these be moved to the end of the list, and are you making a follow-up PR to add `_` prefixes?

File: packages/gestalt/src/TableHeaderCell.js [hunk 9]
```diff
@@ -4,28 +4,49 @@ import cx from 'classnames';
...
+   */
   shouldHaveShadow?: boolean,
+  /**
+   * `colSpan` defines the number of columns a cell should span.
+   */
   colSpan?: number,
+  /**
+   * `rowSpan` defines the number of rows a cell should span.
```
Comment (dangerismycat) [inline packages/gestalt/src/TableHeaderCell.js:24]: ditto here

File: packages/gestalt/src/TableHeaderCell.js [hunk 10]
```diff
@@ -4,28 +4,49 @@ import cx from 'classnames';
...
+   */
   rowSpan?: number,
+  /**
+   * The scope attribute specifies whether a header cell is a header for a column, row, or group of columns or rows. The scope attribute has no visual effect in ordinary web browsers, but can be used by screen readers.
+   */
   scope?: 'col' | 'colgroup' | 'row' | 'rowgroup',
+  /**
+   * Private prop required for sticky columns
```
Comment (dangerismycat) [inline packages/gestalt/src/TableHeaderCell.js:32]: ditto here

File: packages/gestalt/src/TableHeaderCell.js [hunk 11]
```diff
@@ -4,28 +4,49 @@ import cx from 'classnames';
...
-    children,
-    colSpan,
-    scope,
-    rowSpan,
-    shouldBeSticky,
-    previousTotalWidth,
-    shouldHaveShadow,
-  } = props;
+export default function TableHeaderCell({
+  children,
```
Comment (dangerismycat) [inline packages/gestalt/src/TableHeaderCell.js:42]: nit: do you mind alphabetizing these while you're here?

File: packages/gestalt/src/TableSortableHeaderCell.js [hunk 12]
```diff
@@ -6,44 +6,73 @@ import Box from './Box.js';
...
+      | SyntheticMouseEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLAnchorElement>,
+    dangerouslyDisableOnNavigation: () => void,
+  |}) => void,
+  /**
+   * Private prop required for sticky columns
+   */
   previousTotalWidth?: number,
+  /**
+   * `rowSpan` defines the number of rows a cell should span.
```
Comment (dangerismycat) [inline packages/gestalt/src/TableSortableHeaderCell.js:35]: ditto here

**Review by dangerismycat — APPROVED** [review]

File: packages/gestalt/src/Table.js (hunk 1 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Table.js:28]: DONE

File: packages/gestalt/src/Table.js (hunk 2 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Table.js:32]: DONE

File: packages/gestalt/src/Table.js (hunk 3 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Table.js:36]: DONE

File: docs/pages/table.js (hunk 5 above)
Comment (AlbertCarreras) [reply docs/pages/table.js:2574]: DONE

File: packages/gestalt/src/TableHeaderCell.js (hunk 11 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/TableHeaderCell.js:42]: dONE

File: packages/gestalt/src/TableSortableHeaderCell.js [hunk 13]
```diff
@@ -6,44 +6,73 @@ import Box from './Box.js';
...
+  onSortChange: ({|
+    event:
+      | SyntheticMouseEvent<HTMLDivElement>
+      | SyntheticKeyboardEvent<HTMLDivElement>
+      | SyntheticMouseEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLAnchorElement>,
+    dangerouslyDisableOnNavigation: () => void,
+  |}) => void,
+  /**
+   * Private prop required for sticky columns
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply packages/gestalt/src/TableSortableHeaderCell.js:31]: DONE

File: packages/gestalt/src/TableSortableHeaderCell.js (hunk 12 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/TableSortableHeaderCell.js:35]: DONE

File: packages/gestalt/src/TableSortableHeaderCell.js [hunk 14]
```diff
@@ -6,44 +6,73 @@ import Box from './Box.js';
...
+   */
   previousTotalWidth?: number,
+  /**
+   * `rowSpan` defines the number of rows a cell should span.
+   */
   rowSpan?: number,
+  /**
+   * The scope attribute specifies whether a header cell is a header for a column, row, or group of columns or rows. The scope attribute has no visual effect in ordinary web browsers, but can be used by screen readers.
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply packages/gestalt/src/TableSortableHeaderCell.js:39]: DONE

File: packages/gestalt/src/TableRowExpandable.js [hunk 15]
```diff
@@ -1,45 +1,65 @@
...
 import styles from './Table.css';
 import Box from './Box.js';
 import IconButton from './IconButton.js';
 import TableCell from './TableCell.js';
-import { type AbstractEventHandler } from './AbstractEventHandler.js';
 import { useTableContext } from './contexts/TableContext.js';
 
 type Props = {|
+  /**
+   * Supply a short, descriptive label for screen-readers as a text alternative to the Expand button. Accessibility: It populates `aria-label` on the `<button>` element for the Expand button.
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply packages/gestalt/src/TableRowExpandable.js:13]: DONE

File: packages/gestalt/src/TableHeader.js (hunk 6 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/TableHeader.js:12]: DONE

File: packages/gestalt/src/TableHeaderCell.js [hunk 16]
```diff
@@ -4,28 +4,49 @@ import cx from 'classnames';
...
+  /**
+   * `colSpan` defines the number of columns a cell should span.
+   */
   colSpan?: number,
+  /**
+   * `rowSpan` defines the number of rows a cell should span.
+   */
   rowSpan?: number,
+  /**
+   * The scope attribute specifies whether a header cell is a header for a column, row, or group of columns or rows. The scope attribute has no visual effect in ordinary web browsers, but can be used by screen readers.
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply packages/gestalt/src/TableHeaderCell.js:24]: DONE

File: packages/gestalt/src/TableHeaderCell.js (hunk 9 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/TableHeaderCell.js:20]: DONE

**Pull request merged**